### PR TITLE
Add what's new for December

### PR DESCRIPTION
**Pull request by BillWagner**
- Add the what's new article for December 2020.
- Update the TOC for YML validation.

### CODE REVIEW

**Comment by BillWagner** [comment]
@adegeo This doesn't include the `docs-desktop` area yet. If you give a list of what I should find, I'll add the config file and re-run to include those.

File: docs/whats-new/2020-12.md [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+title: ".NET docs: What's new for December 2020"
+description: "What's new in the .NET docs for December 2020."
+ms.date: 01/04/2021
+---
+
+# .NET docs: What's new for December 2020
+
+Welcome to what's new in the .NET docs for December 2020. This article lists some of the major changes to docs during this period.
+
+## .NET
```
Comment (tdykstra) [inline docs/whats-new/2020-12.md:11]: There isn't much for .NET and .NET fundamentals, we could manually combine those two in and .NET Core under one .NET heading

File: docs/whats-new/2020-12.md [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+- [Resolve assembly loads](../standard/assembly/resolve-loads.md) - Assembly loading
+
+## .NET Core
+
+### New articles
+
+- [Debugging StackOverflow errors](../core/diagnostics/debug-stackoverflow.md) - Add debug-stackoverflow.md
+- [SOS debugging extension](../core/diagnostics/sos-debugging-extension.md) - Documentation for SOS
+- [Well-known event providers in .NET](../core/diagnostics/well-known-event-providers.md) - Add docs on well-known event providers
+- [Blazor: Route precedence logic changed in Blazor apps](../core/compatibility/aspnet-core/5.0/blazor-routing-logic-changed.md) - Document Blazor route matching precedence logic
```
Comment (tdykstra) [inline docs/whats-new/2020-12.md:28]: We should find a way to route ASP.NET Core breaking changes to the ASP.NET Core what's new in docs article.

**Review by tdykstra — APPROVED** [review]
A couple minor suggestions, one about tool programming rather than this PR.

File: docs/whats-new/2020-12.md (hunk 2 above)
Comment (BillWagner) [reply docs/whats-new/2020-12.md:28]: Adding @scottaddie 

For the moment, we should just copy & paste, I think.

File: docs/whats-new/2020-12.md (hunk 1 above)
Comment (scottaddie) [reply docs/whats-new/2020-12.md]: Once I ship a new version of the what's new page generator tool, your repo's config file can be updated to automatically merge multiple areas/folders into 1 heading.

**Comment by gewarren** [comment]
Do we want to include the architecture folder in the what's new?

**Pull request merged**